riscv/tests/instruction_tests/generated/lw.S

```diff
@@ -85,30 +85,29 @@
   # Basic tests
   #-------------------------------------------------------------
 
-  test_2: li x10, 2; ebreak; la x1, tdat1; lw x3, 0(x1);; li x29, 0x00ff00ff; li x28, 2; bne x3, x29, fail;;
-  test_3: li x10, 3; ebreak; la x1, tdat2; lw x3, 0(x1);; li x29, 0xff00ff00; li x28, 3; bne x3, x29, fail;;
-  test_4: li x10, 4; ebreak; la x1, tdat3; lw x3, 0(x1);; li x29, 0x0ff00ff0; li x28, 4; bne x3, x29, fail;;
-  test_5: li x10, 5; ebreak; la x1, tdat4; lw x3, 0(x1);; li x29, 0xf00ff00f; li x28, 5; bne x3, x29, fail;;
+  test_2: li x10, 2; ebreak; la x1, tdat; lw x3, 0(x1);; li x29, 0x00ff00ff; li x28, 2; bne x3, x29, fail;;
+  test_3: li x10, 3; ebreak; la x1, tdat; lw x3, 4(x1);; li x29, 0xff00ff00; li x28, 3; bne x3, x29, fail;;
+  test_4: li x10, 4; ebreak; la x1, tdat; lw x3, 8(x1);; li x29, 0x0ff00ff0; li x28, 4; bne x3, x29, fail;;
+  test_5: li x10, 5; ebreak; la x1, tdat; lw x3, 12(x1);; li x29, 0xf00ff00f; li x28, 5; bne x3, x29, fail;;
 
   # Test with negative offset
 
-  test_6: li x10, 6; ebreak; la x1, tdat1; lw x3, 0(x1);; li x29, 0x00ff00ff; li x28, 6; bne x3, x29, fail;;
-  test_7: li x10, 7; ebreak; la x1, tdat2; lw x3, 0(x1);; li x29, 0xff00ff00; li x28, 7; bne x3, x29, fail;;
-  test_8: li x10, 8; ebreak; la x1, tdat3; lw x3, 0(x1);; li x29, 0x0ff00ff0; li x28, 8; bne x3, x29, fail;;
+  test_6: li x10, 6; ebreak; la x1, tdat4; lw x3, -12(x1);; li x29, 0x00ff00ff; li x28, 6; bne x3, x29, fail;;
+  test_7: li x10, 7; ebreak; la x1, tdat4; lw x3, -8(x1);; li x29, 0xff00ff00; li x28, 7; bne x3, x29, fail;;
+  test_8: li x10, 8; ebreak; la x1, tdat4; lw x3, -4(x1);; li x29, 0x0ff00ff0; li x28, 8; bne x3, x29, fail;;
   test_9: li x10, 9; ebreak; la x1, tdat4; lw x3, 0(x1);; li x29, 0xf00ff00f; li x28, 9; bne x3, x29, fail;;
 
   # Test with a negative base
 
-  test_10: li x10, 10; ebreak; la x1, tdat1; addi x1, x1, -32; lw x3, 32(x1);; li x29, 0x00ff00ff; li x28, 10; bne x3, x29, fail;
+  test_10: li x10, 10; ebreak; la x1, tdat; addi x1, x1, -32; lw x3, 32(x1);; li x29, 0x00ff00ff; li x28, 10; bne x3, x29, fail;
 
 
 
 
 
   # Test with unaligned base
 
-  # TODO enable it when we support unaligned memory
-  #test_11: li x10, 11; ebreak; la x1, tdat; addi x1, x1, -3; lw x3, 7(x1);; li x29, 0xff00ff00; li x28, 11; bne x3, x29, fail;
+  test_11: li x10, 11; ebreak; la x1, tdat; addi x1, x1, -3; lw x3, 7(x1);; li x29, 0xff00ff00; li x28, 11; bne x3, x29, fail;
```

Comment:
Does this work like that?

Reply:
Yes. I forgot why I thought it would work, but it does.